0000-interledger-protocol-4/0000-interledger-protocol-4.md

```diff
@@ -0,0 +1,146 @@
+
+ILPv4 will probably be used with a Transport Layer protocol such as the Pre-Shared Key V2 (PSKv2) protocol, which handles the generation of ILP conditions and fulfillments.
+
+### Differences from Previous Versions of ILP
+
+- **Designed for Smaller, More Homogenous Packet Amounts** - ILPv4 applies a number of simplifications based on the change from supporting all packet amounts to focusing the Interledger layer on lower-value packets. Larger amounts can be sent using chunked payments, but the core network is optimized for sending large volumes of small packets. This renders unnecessary some of the more complex features of previous versions like Liquidity Curves for expressing how exchange rates may vary depending on the packet amount. Lower-value packets also help to minimize [connector risks](../0018-connector-risk-mitigations/0018-connector-risk-mitigations.md).
+- **Payment Channels, Not On-Ledger Escrow** - Since ILPv4 is optimized for smaller packets, speed and cost are of greater importantance than in previous versions. ILPv4 uses ledgers or payment channels for settling bilateral payment obligations but ILPv4 packets are sent just between connectors, rather than through the underlying ledgers themselves. This enables packet timeouts to be short, because they do not need to include the processing time of slower ledgers, which further reduces [connector risks](../0018-connector-risk-mitigations/0018-connector-risk-mitigations.md).
+- **Forwarding, Not Delivery** - In contrast to the first version of ILP, ILPv4 does not have the destination amount in the packet and connectors must only apply their local rate to forward packets. Senders may use higher-level protocols to indicate to the receiver the minimum amount they should accept for each packet, but this is no longer built into the core protocol. Connectors have only one simple behavior and do not need to maintain up-to-date price information on all possible destintations.
+- **Quoting is an Application Concern** - Connectors are only responsible for forwarding ILP packets and do not need to implement a [separate protocol for quoting](../0008-interledger-quoting-protocol/0008-interledger-quoting-protocol.md). Applications can use test packets to determine the exchange rate of a particular path.
+
+## Flow
+
+Before the protocol starts, the sender and receiver use a higher-level protocol to agree on the destination account and packet condition.
+
+1. Sender constructs an ILP Prepare packet with the receiver's destination account, the agreed-upon condition, and an amount and expiry of their choice. The sender may include additional data, the formatting of which is determined by the higher-level protocol.
```

Comment:
What is the difference? The expiry / deadline by which each bilateral transfer must be made or rolled back decreases with each successive hop so that the connectors have a window of time to deliver the fulfillment back even if the next transfer executes at the last possible moment.

Reply:
It's a question about whether it's represented as a point in time ("February 9th, at 17:24 UTC") or a duration of time ("3600 seconds from now").

You can't "shorten" a point, so ultimately this may just be a nitpick with that wording.

If it's a duration, then that raises other questions. (Whose "now" are we using? Which end do I shorten it from? etc.)

Reply:
It's a timestamp, so I'll adjust the wording about it

Reply:
I second others opinion, that it is much more elegant and simplified protocol. also the naming conventions are very well placed for comparison with tradition ipv4 & hence the understanding of layers/packets. I have two minor points that I found was not clear to me.

1.) using PaymentChannels instead of hashlocks - I understand they do not need to include the processing time of slower ledgers, hence connector risks & somewhat even underlying ledger (architectural) issues are mitigated. 

However How can this produce the same or better settlement of Bilateral payment obligations ? 

2.) For proof of settlement - the document says ".. sender's signed payment channel update (or other proof of settlement) .. " 
But could not find what other proofs of settlement are accepted and is this something that a user/entity can change ( ex: add/select ones own accepted proofs among a bunch of  proofs that are "verified for use"? )

As i mentioned, really find this a much more simplified & way more elegant .  I think its a great move to have such close design co-relation with the ip stack. Thanks for all the great work kindly.